### PR TITLE
doc: add a section  about debugging in docker-compose with PyCharm

### PR DESCRIPTION
**Pull request by WackyGem**
Proposed Documentation Pull Request for Airflow

Background:
In a discussion with @potiuk about [Seeking Help: Missing Essential Python Packages during PyCharm Debugging on Docker Compose Interpreter](https://github.com/apache/airflow/discussions/37864). I proposed adding configuration instructions for this functionality to the Airflow documentation.

Changes:
This pull request adds a new section to the documentation that covers:
**Debug Airflow inside docker container using Pycharm**

I would appreciate it if you could review this pull request and provide any feedback.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by vincbeck — APPROVED** [review]

File: docs/apache-airflow/howto/docker-compose/index.rst [hunk 1]
```diff
@@ -358,6 +358,52 @@ Networking
...
+===================================================
+Prerequisites: Create a project in **PyCharm** and download the `docker-compose.yaml <{{ doc_root_url }}docker-compose.yaml>`__.
+
+Steps:
+
+1) Modify  ``docker-compose.yaml``
+
+   Add the following section under the ``services`` section:
+
+.. code-block:: docker
```
Comment (Taragolis) [inline docs/apache-airflow/howto/docker-compose/index.rst]:
```suggestion
.. code-block:: yaml
```

File: docs/apache-airflow/howto/docker-compose/index.rst [hunk 2]
```diff
@@ -358,6 +358,52 @@ Networking
...
+
+    services:
```
Comment (Taragolis) [inline docs/apache-airflow/howto/docker-compose/index.rst]: I'm sure that I know that this line shouldn't be copied.
But worried that end users copy it under services with preserving this line, e.g.

```yaml
services:
  ..
  services:
      airflow-python:
        <<: *airflow-common
        profiles:
          - debug
        environment:
          <<: *airflow-common-env
        user: "50000:0"
        entrypoint: ["bash"]
```

File: docs/apache-airflow/howto/docker-compose/index.rst [hunk 3]
```diff
@@ -358,6 +358,52 @@ Networking
 
 In general, if you want to use Airflow locally, your DAGs may try to connect to servers which are running on the host. In order to achieve that, an extra configuration must be added in ``docker-compose.yaml``. For example, on Linux the configuration must be in the section ``services: airflow-worker`` adding ``extra_hosts: - "host.docker.internal:host-gateway"``; and use ``host.docker.internal`` instead of ``localhost``. This configuration vary in different platforms. Please check the Docker documentation for `Windows <https://docs.docker.com/desktop/windows/networking/#use-cases-and-workarounds>`_ and `Mac <https://docs.docker.com/desktop/mac/networking/#use-cases-and-workarounds>`_ for further information.
 
+Debug Airflow inside docker container using Pycharm
+===================================================
+Prerequisites: Create a project in **PyCharm** and download the `docker-compose.yaml <{{ doc_root_url }}docker-compose.yaml>`__.
```
Comment (Taragolis) [inline docs/apache-airflow/howto/docker-compose/index.rst]: I'm not sure why but locally URI doesn't rendered correctly: `http://127.0.0.1:8000/docs/apache-airflow/stable/howto/docker-compose/%7B%7Bdoc_root_url%7D%7Ddocker-compose.yaml`

File: docs/apache-airflow/howto/docker-compose/index.rst (hunk 1 above)
Comment (WackyGem) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I'll make the changes you suggested.

File: docs/apache-airflow/howto/docker-compose/index.rst (hunk 2 above)
Comment (WackyGem) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I'll make the changes you suggested.

File: docs/apache-airflow/howto/docker-compose/index.rst (hunk 3 above)
Comment (WackyGem) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I copy `docker-compose.yaml <{{ doc_root_url }}docker-compose.yaml>`__. from https://github.com/apache/airflow/blob/ca11b7af239c90f6d7c00ed5116fdd08fda1a5ab/docs/apache-airflow/howto/docker-compose/index.rst?plain=1#L77
I'm not sure why it's happening.

Comment (Taragolis) [reply docs/apache-airflow/howto/docker-compose/index.rst]: Some magic of jinja templating doesn't work with sphinx 🤔 It works well before in other place, however in this place it keep unchanged even thought it exatly the same as it defined in `Fetching docker-compose.yaml`

Comment (Taragolis) [reply docs/apache-airflow/howto/docker-compose/index.rst]: Ahhhh... figure it out

```suggestion
.. jinja:: quick_start_ctx

    Prerequisites: Create a project in **PyCharm** and download the `docker-compose.yaml <{{ doc_root_url }}docker-compose.yaml>`__.

```

image

Comment (WackyGem) [reply docs/apache-airflow/howto/docker-compose/index.rst]: OH I see, this template only works in the Jinja context. I will add ``.. jinja:: quick_start_ctx``.

Comment (WackyGem) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I will work on these changes in my spare time, including the parts that failed the static checks. However, it is late at night for me, so I need to rest first. You can go ahead and work on other things.

Comment (Taragolis) [reply docs/apache-airflow/howto/docker-compose/index.rst]: No worries, there is no rush required

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice!

**Pull request merged**